### DISCUSSION TITLE
Userdefaults의 변화를 자동으로 UI를 변화하게 만들고 싶어요

### DISCUSSION THREAD

**Discussion by seongahshin** [DeveloperAcademy-POSTECH/.github#26] [Q&A]
### 현상 
Userdefaults 가 변화면 자동으로 UI까지 변화하게 만들고 싶어요... 

```swift
if product.displayName == "10장만 보기" {
    UserDefaults.standard.tenPhotosAccess = true
} else {
    UserDefaults.standard.allPhotosAccess = true
}
```

### 원인 
Usedefaults 에는 자체적으로 변화를 감지할 수 있는 기능이 존재하지 않습니다.. 그래서 ObservableObject Class 로 감싸서 Published 를 사용해야 하는지.. 아니면 또 다른 좋은 방법이 있는 것인지 모르겠습니다!

**Comment by moonkey48** [comment]
SwiftUI에서는 UserDefault를 핸들링하기 위해 AppStorage라는 프로퍼티래퍼가 있습니다 :)

기본 형식은 아래와 같습니다. 
```swift
AppStorage("userDefaultKeyName") var userDefaultVariable: Bool = false
```

> 위 코드에서 userDefaultKeyName는 UserDefault의 key값이고 뒤에 있는 userDefaultVariable는 해당 키값에 대한 UserDefault의 값을 가져와서 저장합니다. 뒤에는 기본값이 할당되어 있는데 그 전에 할당된 값이 있는 경우에는 해당 값을 사용하고, 한번도 해당 키에 대한 값이 할당되지 않은 경우에는 `= false`와 같이 기본값을 할당해줄 수 있습니다. AppStorage변화에 따라 뷰는 자동으로 업데이트가 이루어지게 됩니다:)


https://developer.apple.com/documentation/swiftui/appstorage
